modules/core/src/test/java/org/apache/synapse/mediators/transform/pfutils/FreeMarkerTemplateProcessorTest.java

```diff
@@ -421,6 +422,54 @@ public void testWithXMLPayloadWithFtlHeader() throws Exception {
                 + "set expected format", expectedEnvelopeFtl, synCtx.getEnvelope().getBody().toString());
     }
 
+    public void testMultipleFtlPayloads() throws Exception {
```

Comment:
We need to have an integration test to validate this fix. Since you have explicitly called `init()` in the two templateProcessor instances, this will work even without the fix. 

We need to write a synapse sequence with two Payload factory mediators and invoke it and verify. Please refer https://github.com/wso2/wso2-synapse/pull/1739/files to get an idea. If you need further clarifications let's have an offline discussion.

Reply:
Moving this test case to micro integrator from synapse